modules/loader/src/main/java/com/opengamma/strata/loader/csv/IborCapFloorTradeCsvPlugin.java

```diff
@@ -0,0 +1,330 @@
+    if (index instanceof IborIndex) {
+      return (IborIndex) index;
+    } else {
+      throw new IllegalArgumentException("Index " + index.getName() + "is not an IBOR index");
+    }
+  }
+
+  // parse the periodic schedule for CapFloor
+  private static PeriodicSchedule parseSchedule(CsvRow row, Currency currency) {
+    PeriodicSchedule.Builder builder = PeriodicSchedule.builder();
+    // basics
+    builder.startDate(row.getValue(START_DATE_FIELD, LoaderUtils::parseDate));
+    builder.endDate(row.getValue(END_DATE_FIELD, LoaderUtils::parseDate));
+    builder.frequency(Frequency.parse(row.getValue(PAYMENT_FREQUENCY_FIELD)));
+    // stub period is not allowed for CapFloor trade
+    builder.stubConvention(StubConvention.NONE);
+    // adjustments
+    BusinessDayAdjustment dateAdj =
+        CsvLoaderUtils.parseBusinessDayAdjustment(row, DATE_ADJ_CNV_FIELD, DATE_ADJ_CAL_FIELD)
+            .orElse(BusinessDayAdjustment.NONE);
+    builder.businessDayAdjustment(dateAdj);
+    CsvLoaderUtils.parseBusinessDayAdjustment(row, START_DATE_CNV_FIELD, START_DATE_CAL_FIELD)
+        .ifPresent(builder::startDateBusinessDayAdjustment);
+    CsvLoaderUtils.parseBusinessDayAdjustment(row, END_DATE_CNV_FIELD, END_DATE_CAL_FIELD)
+        .ifPresent(builder::endDateBusinessDayAdjustment);
+    // optionals
+    row.findValue(ROLL_CONVENTION_FIELD)
+        .map(LoaderUtils::parseRollConvention)
+        .ifPresent(builder::rollConvention);
+    row.findValue(FIRST_REGULAR_START_DATE_FIELD)
+        .map(LoaderUtils::parseDate)
+        .ifPresent(builder::firstRegularStartDate);
+    row.findValue(LAST_REGULAR_END_DATE_FIELD)
+        .map(LoaderUtils::parseDate)
+        .ifPresent(builder::lastRegularEndDate);
+    Optional<AdjustableDate> overrideDateOpt = row.findValue(OVERRIDE_START_DATE_FIELD)
+        .map(ignored -> CsvLoaderUtils.parseAdjustableDate(
+            row,
+            OVERRIDE_START_DATE_FIELD,
+            OVERRIDE_START_DATE_CNV_FIELD,
+            OVERRIDE_START_DATE_CAL_FIELD,
+            BusinessDayConventions.MODIFIED_FOLLOWING,
+            currency));
+    overrideDateOpt.ifPresent(builder::overrideStartDate);
+    return builder.build();
+  }
+
+  // parse the upfront premium amount
+  private static Optional<AdjustablePayment> parsePremium(CsvRow row) {
+    Optional<Double> premiumAmountOpt = row.findValue(PREMIUM_AMOUNT_FIELD)
+        .map(LoaderUtils::parseDouble);
+    if (premiumAmountOpt.isPresent()) {
+      Currency premiumCurrency = row.getValue(PREMIUM_CURRENCY_FIELD, LoaderUtils::parseCurrency);
+      PayReceive premiumDirection = row.getValue(PREMIUM_DIRECTION_FIELD, LoaderUtils::parsePayReceive);
+      LocalDate premiumDate = row.getValue(PREMIUM_DATE_FIELD, LoaderUtils::parseDate);
+      CurrencyAmount premiumAmount = CurrencyAmount.of(premiumCurrency, premiumAmountOpt.get());
+      AdjustablePayment payment = premiumDirection.isPay() ? AdjustablePayment.ofPay(premiumAmount, premiumDate)
```

Comment:
When breaking a tertiary onto multiple lines there should be a new line after the ? and after the :